core/interval/int.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package interval
```

Comment:
I would rename this package, it has one specific need. Also, I would de-generalise it, name things what they are. I say this, because people might believe this does something that it shouldn't. If people want a generalised interval with grouped sets, then they can copy it and do it how they want.

Reply:
I was beginning to come to this conclusion responding to some earlier comments. Refactoring away I go I guess!

Instead of using the general interval concept, I have concretised around a new `PortRanges` type in `core/network/portrange.go`, aliased around a slice of `PortRanges`.

I have given this type a constructor, and methods to Add and Remove port ranges, merging as we go.

This is quite nice because it allows me to encapsulate the special treatments of 'icmp' closer together